ionoscloud/resource_dbaas_redisdb_replicaset.go

```diff
@@ -0,0 +1,291 @@
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"maintenance_window": {
+				Type:        schema.TypeList,
+				Description: "A weekly 4 hour-long window, during which maintenance might occur.",
+				Optional:    true,
+				Computed:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:        schema.TypeString,
+							Description: "Start of the maintenance window in UTC time.",
+							Required:    true,
+						},
+						"day_of_the_week": {
+							Type:        schema.TypeString,
+							Description: "The name of the week day.",
+							Required:    true,
+						},
+					},
+				},
+			},
+			"initial_snapshot_id": {
+				Type:        schema.TypeString,
+				Description: "The ID of a snapshot to restore the replica set from. If set, the replica set will be created from the snapshot.",
+				Optional:    true,
+			},
+			"dns_name": {
+				Type:        schema.TypeString,
+				Description: "The DNS name pointing to your replica set. Will be used to connect to the active/standalone instance.",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func redisDBReplicaSetCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).RedisDBClient
+
+	replicaSet := redisdb.GetRedisDBReplicaSetDataCreate(d)
+	response, _, err := client.CreateRedisDPReplicaSet(ctx, replicaSet, d.Get("location").(string))
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occurred while creating a DBaaS RedisDB replica set: %w", err))
+	}
+	replicaSetID := *response.Id
+	d.SetId(replicaSetID)
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsReplicaSetReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error occurred while checking the status for RedisDB replica set with ID: %v, error: %w", replicaSetID, err))
+	}
+	// Call the read function to save the DNS name in the state (DNS name is not present in the create response).
+	return redisDBReplicaSetRead(ctx, d, meta)
+}
+
+func redisDBReplicaSetDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).RedisDBClient
+	replicaSetID := d.Id()
+	apiResponse, err := client.DeleteRedisDBReplicaSet(ctx, replicaSetID, d.Get("location").(string))
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error while deleting RedisDB replica set with ID: %v, error: %w", replicaSetID, err))
+	}
+	err = utils.WaitForResourceToBeDeleted(ctx, d, client.IsReplicaSetDeleted)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("deletion check failed for RedisDB replica set with ID: %v, error: %w", replicaSetID, err))
+	}
+
+	// wait for the lan to be freed after the deletion of the replica set
+	time.Sleep(constant.SleepInterval * 10)
```

Comment:
do we need this sleep?

Reply:
@cristiGuranIonos not sure yet, we will find out when the API team will solve the LAN deletion problem.